Feature : Support LTR / RTL CSS Styling

anbinm-msft opened the pull request.

msftclas commented:
Hi __@mttwc__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no Contribution License Agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, MSBOT;

leakymemory committed:
Clean slate

leakymemory committed:
Baseline for React

leakymemory committed:
Minor updates to generate shared .css

leakymemory committed:
Put the basic component structure in place

leakymemory committed:
Add basic expand/collapse

leakymemory committed:
Merge pull request #16 from OneNoteDev/feature/collapse-tree

Add basic expand/collapse

mttwc committed:
Added a global props and an interface to handle OneNote data structure related side effects

mttwc committed:
Deprecate the data source and use the new interface instead

leakymemory committed:
First pass at style updates

This is only the first pass, and is not according to redlines yet.  One of
the more significant changes here is to switch over to use the Office
UI Fabric styles.  This is something we're going to need to pull into the
component itself (vs. require it to be included outside).

leakymemory committed:
Switch the background color to white.

mttwc committed:
Implement side effects, with a preliminary mocked api call to get pages on section click

leakymemory committed:
Merge pull request #17 from OneNoteDev/feature/styling-update

Feature/styling update

anbiniyar committed:
Ignore IDEA specific files

anbiniyar committed:
Enforce whitespace standards across IDEs

anbiniyar committed:
Add karma + enzyme with code coverage

anbiniyar committed:
Add WallabyJS Support

anbiniyar committed:
Ensure .ts and .tsx is covered in the tests

anbiniyar committed:
Rename ‘spec’ folder to ‘test’ to add clarity

anbiniyar committed:
Add newline at the end of the file

mttwc committed:
Merge branch 'feature/move-to-react' into feature/side-effect-handling

mttwc committed:
Merge pull request #19 from OneNoteDev/feature/move-to-react-add-karma

Add testing support to Project

mttwc committed:
Merge branch 'feature/move-to-react' into feature/side-effect-handling

mttwc committed:
Added sinon and the ability to mock global props

mttwc committed:
Added some tests for NotebookListUpdater

mttwc committed:
Merge pull request #18 from OneNoteDev/feature/side-effect-handling

Feature/side effect handling

mttwc committed:
Add page updating tests to NotebookListUpdater tests

mttwc committed:
Naive implementation for the OneNoteApiDataProvider

mttwc committed:
Bubble up the selected section, and use that to apply styling in the picker

mttwc committed:
Merge pull request #21 from OneNoteDev/feature/selection-logic

Selection logic and styling

mttwc committed:
Merge pull request #20 from OneNoteDev/feature/side-effect-handling-tests

Feature/side effect handling tests

mttwc committed:
Add the ability for the dev to specify what is selectable

mttwc committed:
Convert spaces to tabs

mttwc committed:
Merge pull request #23 from OneNoteDev/feature/custom-selections-and-expands

Feature/custom selections and expands

mttwc committed:
Only show expandable chevron if item is expandable

mttwc committed:
Merge pull request #24 from OneNoteDev/feature/only-show-chevron-on-expandable

Only show expandable chevron if item is expandable

mttwc committed:
Remove carets. Styled font, margins, and removed borders.

leakymemory committed:
First pass at TSLint updates

leakymemory committed:
Merge pull request #30 from OneNoteDev/feature/tslint-updates

First pass at TSLint updates

leakymemory committed:
Package updates

mttwc committed:
Style to redlines

mttwc committed:
Remove selectability on icons

mttwc committed:
Style the dimensions of rows and entire picker sample to redlines

mttwc committed:
Add aria accessibility

mttwc committed:
Add tabbing

mttwc committed:
Add polyfills for promise and find

mttwc committed:
Use es6-shim for typing and es6-promise to polyfill the promise

mttwc committed:
Merge pull request #34 from OneNoteDev/feature/accessibility

Feature/accessibility

mttwc committed:
Merge branch 'feature/move-to-react' into feature/style-to-redlines

mttwc committed:
Merge pull request #31 from OneNoteDev/feature/style-to-redlines

Feature/style to redlines

mttwc committed:
Merge pull request #33 from OneNoteDev/feature/polyfills

Add polyfills for promise and find

mttwc committed:
Add parent ref to each onenote item.

mttwc committed:
Create a child item interface to abstract away some common code from our onenote data structures

mttwc committed:
Some refactoring around onenote items. Added a simple util to get breadcrumbs, which still needs to be tested

mttwc committed:
Test breadcrumbs, fix bug in the way API objects were being converted into POJOs

mttwc committed:
Clearer console log debugging of breadcrumbs

anbinm-msft committed:
Prepare webpack config for production release

- Ignore dist folder in tsconfig
- Import style into the tsx file so webpack can generate .css .js with
  same name
- Clean the dist folder before generating prod file
- Separate dev config from prod config
- Remove commented code

leakymemory committed:
Minor package updates.

mttwc committed:
Merge pull request #37 from OneNoteDev/feature/robust-notebook-hierarchy

Feature/robust notebook hierarchy

anbinm-msft committed:
Add production build and bundle analyzer

- Generate .js and .min.js files
- Add support for bundle analyzer

leakymemory committed:
Namespace the component CSS

We want to make sure that the CSS provided with the component doesn't conflict
with any of the other page elements.

leakymemory committed:
Fix the indent.

leakymemory committed:
Merge pull request #40 from OneNoteDev/bug/namespace-the-css

Bug/namespace the css

anbinm-msft committed:
Add PostCSS loader for auto prefixing

- Remove any form of minification

anbinm-msft committed:
Simplify production configuration

- This runs webpack twice so we can produce minified and non-minified
  version easily.

anbinm-msft committed:
Create a standalone example to consume picker

Make sure we are exporting react correctly

anbinm-msft committed:
Merge branch 'feature/move-to-react' into move-to-react-production-build

leakymemory committed:
Merge pull request #41 from OneNoteDev/move-to-react-production-build

Produce production version for library

anbiniyar committed:
Add support for LTR / RTL css styling

anbiniyar committed:
Fix inverted naming

This fixes the bug where.min.css was in fact not minified and .css was minified.

msftclas commented:
@anbinm-msft,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_
